### PR TITLE
USHIFT-2239: Destroy VM when virt-install exits with errors

### PR DESCRIPTION
**Pull request by ggiguash**

### CODE REVIEW

**Comment by ggiguash** [comment]
Ready for review. The failure is not related to the change.

File: test/bin/scenario.sh [hunk 1]
```diff
@@ -486,6 +486,10 @@ launch_vm() {
     if ${vm_created} ; then
         record_junit "${vmname}" "install_vm" "OK"
     else
+        # Make sure to stop the VM on error before the control is returned.
+        # This is necessary not to leave running qemu child processes so that
+        # the caller considers the script fully complete.
+        sudo virsh destroy "${full_vmname}" || true
```
Comment (dhellmann) [inline test/bin/scenario.sh:492]: This is going to make it more difficult to debug failures when running scenarios locally. Could we have a flag to control the behavior? Leaving the default set to cleanup automatically would make sense.

Comment (ggiguash) [reply test/bin/scenario.sh]: I only enable this option when stdin is not available (non-interactive session).
Is this better now?

Comment (dhellmann) [reply test/bin/scenario.sh]: That works, thanks!

**Review by pacevedom — COMMENTED** [review]
/lgtm

**Comment by ggiguash** [comment]
FIPS failures are not related to the current change
/override ci/prow/metal-periodic-test

**Pull request merged**